app/v1/controllers/ListPaymentsController.scala

```diff
@@ -70,9 +81,31 @@ class ListPaymentsController @Inject()(val authService: EnrolmentsAuthService,
         val correlationId = getCorrelationId(errorWrapper)
         val result = errorResult(errorWrapper).withApiHeaders(correlationId)
 
+        auditSubmission(
+          AuditDetail(
+            userDetails = request.userDetails,
+            nino = nino,
+            `X-CorrelationId` = correlationId,
+            response = AuditResponse(httpStatus = result.header.status, response = Left(errorWrapper.auditErrors))
+          )
+        )
+
         result
       }.merge
   }
+  private def auditSubmission(details: AuditDetail)
+                             (implicit hc: HeaderCarrier,
+                              ec: ExecutionContext): Future[AuditResult] = {
+
+    val event = AuditEvent(
+      auditType = "retrieveSelfAssessmentTransactions",
```

Comment:
check the audit samples for the list payments in the confluence page and change **auditType** and ** transactionName** accordingly